python/ray/monitor.py

```diff
@@ -142,8 +171,10 @@ def update_load_metrics(self):
                 resources_batch_data.placement_group_load.placement_group_data)
 
             # Update the load metrics for this raylet.
-            node_id = ray.utils.binary_to_hex(resource_message.node_id)
-            ip = self.raylet_id_to_ip_map.get(node_id)
+            node_id = binary_to_hex(resource_message.node_id)
+            # Make sure the node_id size never changes for backward compat.
+            assert len(resource_message.node_id) == 28
```

Comment:
Please remove this assert, it prevents future changes.

Reply:
Why future changes suffer here? If a future change is necessary we can modify it?

Reply:
why is `node_id` here at all? we should just get rid of it and any references to it (including the entire raylet map)